### DISCUSSION TITLE
Why request the production.cloudflare.docker.com ?

### DISCUSSION THREAD

**Discussion by seepine** [docker/buildx#2661] [Q&A]
The workflow yml
```yml
      - name: Build and push
        uses: docker/build-push-action@v4
        with:
          context: .
          file: ./Dockerfile
          platforms: |
            linux/amd64
            linux/arm64
          push: true
          tags: test:latest
```

And got the error
```
 > [linu***/arm64 internal] load metadata for docker.io/library/alpine:latest:
------
Dockerfile:1
--------------------
   1 | >>> FROM alpine
   2 |     CMD echo 'test docker build***!!'
--------------------
ERROR: failed to solve: docker.io/library/alpine: failed to resolve source metadata for docker.io/library/alpine:latest: failed to copy: httpReadSeeker: failed open: failed to do request: Get "https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/1d/1d34ffe**190be23d3de5**de0a436676**58f48f835c3a2d**68b798c15a7f1/data?verify=1718641860-garIfCZ**RLgwzS**JFgHb**0VE%3D": read tcp 172.17.0.3:42364->1*7.2*0.8.36:443: read: connection reset by peer
```

I am using Gitea with Gitea Action in a local network environment. And i have configured a registry mirror, run `docker pull alpine` is able to fetch the image.
Why does the `docker/build-push-action@v4` action need to request from production.cloudflare.docker.com, this leads to build failure.

**Comment by seepine** [comment]
### Q1: Why request the production.cloudflare.docker.com

I config `/etc/docker/daemon.json` of `registry-mirrors` and run `docker pull nginx:1.25-alpine-slim` can work. But docker buildx build throw error.
```
ERROR: failed to solve: nginx:1.25-alpine-slim: failed to resolve source metadata for docker.io/library/nginx:1.25-alpine-slim: failed to copy: httpReadSeeker: failed open: failed to do request: Get "https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/2c/2c84e1e0****5023d558e/data?verify=1723****dPXX9c%3D": dial tcp 31.13.95.48:443: connect: connection refused
```

### Q2: Why cannot use docker proxy ?
I set empty `/etc/docker/daemon.json` of `registry-mirrors`, and config `/etc/systemd/system/docker.service.d/proxy.conf` like this
```
[Service]
Environment="HTTP_PROXY=http://192.168.1.1:7890/"
Environment="HTTPS_PROXY=http://192.168.1.1:7890/"
Environment="NO_PROXY=localhost,127.0.0.1"
```
And run  `docker pull nginx:1.25-alpine-slim` can work. But docker buildx build throw error also.

**Reply by snowdream** [reply]
I think i get the same error.

https://github.com/docker/buildx/issues/2667
https://github.com/docker/build-push-action/issues/1223
